stdlib/stream.ml

```diff
@@ -15,14 +15,15 @@
    the empty stream. This is type safe because the empty stream is never
    patched. *)
 
-type 'a t = { count : int; data : 'a data }
+type 'a t = 'a data' option
```

Comment:
`data'` is really hard to read and visually distinguish from `data` as a datatype name. Could you use `cell` instead, or whatever?